*** ./Sources/top/psl_fpga.vhdl	2016-11-30 19:21:08.000000002 +0100
--- ./Sources/top/psl_fpga.vhdl	2017-01-24 18:31:41.000000002 +0100
***************
*** 66,73 ****
         pci0_o_txn_out6: out std_logic;
         pci0_o_txp_out7: out std_logic;
         pci0_o_txn_out7: out std_logic;
!        o_debug: out std_logic_vector(0 to 3));
  
         attribute secure_config : string;
         attribute secure_config of psl_fpga : entity is "PROTECT";
         attribute secure_netlist : string;       
--- 66,96 ----
         pci0_o_txn_out6: out std_logic;
         pci0_o_txp_out7: out std_logic;
         pci0_o_txn_out7: out std_logic;
!        o_debug: out std_logic_vector(0 to 3);
  
+        -- FlashGT i/o for DDR4 SDRAM
+        refclk200_p        : IN    STD_LOGIC;
+        refclk200_n        : IN    STD_LOGIC;
+        c0_sys_clk_p       : IN    STD_LOGIC;
+        c0_sys_clk_n       : IN    STD_LOGIC;
+ -- TODO:       dbg_bus            : OUT   STD_LOGIC_VECTOR(511 DOWNTO 0);
+        c0_ddr4_adr        : OUT   STD_LOGIC_VECTOR(16 DOWNTO 0);
+        c0_ddr4_ba         : OUT   STD_LOGIC_VECTOR(1 DOWNTO 0);
+        c0_ddr4_cke        : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0);
+        c0_ddr4_cs_n       : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0);
+        c0_ddr4_dm_dbi_n_9 : INOUT STD_LOGIC;
+        c0_ddr4_dm_dbi_n   : INOUT STD_LOGIC_VECTOR(8 DOWNTO 0);
+        c0_ddr4_dq         : INOUT STD_LOGIC_VECTOR(71 DOWNTO 0);
+        c0_ddr4_dqs_c      : INOUT STD_LOGIC_VECTOR(8 DOWNTO 0);
+        c0_ddr4_dqs_t      : INOUT STD_LOGIC_VECTOR(8 DOWNTO 0);
+        c0_ddr4_odt        : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0);
+        c0_ddr4_bg         : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0);
+        c0_ddr4_reset_n    : OUT   STD_LOGIC;
+        c0_ddr4_act_n      : OUT   STD_LOGIC;
+        c0_ddr4_ck_c       : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0);
+        c0_ddr4_ck_t       : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0);
+        c0_ddr4_ten        : OUT   STD_LOGIC
+       );
         attribute secure_config : string;
         attribute secure_config of psl_fpga : entity is "PROTECT";
         attribute secure_netlist : string;       
***************
*** 150,156 ****
         ha_jeapar: in std_ulogic;
         ha_jcompar: in std_ulogic;
         ah_paren: out std_ulogic;
!        ha_pclock: in std_ulogic);
  End Component psl_accel;
  
  Component base_img
--- 173,201 ----
         ha_jeapar: in std_ulogic;
         ha_jcompar: in std_ulogic;
         ah_paren: out std_ulogic;
!        ha_pclock: in std_ulogic;
!
!        -- FlashGT i/o for DDR4 SDRAM
!        refclk200_p      : IN    STD_LOGIC;
!        refclk200_n      : IN    STD_LOGIC;
!        c0_sys_clk_p     : IN    STD_LOGIC;
!        c0_sys_clk_n     : IN    STD_LOGIC;
! -- TODO:       dbg_bus          : OUT   STD_LOGIC_VECTOR(511 DOWNTO 0);
!        c0_ddr4_adr      : OUT   STD_LOGIC_VECTOR(16 DOWNTO 0);
!        c0_ddr4_ba       : OUT   STD_LOGIC_VECTOR(1 DOWNTO 0);
!        c0_ddr4_cke      : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0);
!        c0_ddr4_cs_n     : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0);
!        c0_ddr4_dm_dbi_n : INOUT STD_LOGIC_VECTOR(8 DOWNTO 0);
!        c0_ddr4_dq       : INOUT STD_LOGIC_VECTOR(71 DOWNTO 0);
!        c0_ddr4_dqs_c    : INOUT STD_LOGIC_VECTOR(8 DOWNTO 0);
!        c0_ddr4_dqs_t    : INOUT STD_LOGIC_VECTOR(8 DOWNTO 0);
!        c0_ddr4_odt      : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0);
!        c0_ddr4_bg       : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0);
!        c0_ddr4_reset_n  : OUT   STD_LOGIC;
!        c0_ddr4_act_n    : OUT   STD_LOGIC;
!        c0_ddr4_ck_c     : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0);
!        c0_ddr4_ck_t     : OUT   STD_LOGIC_VECTOR(0 DOWNTO 0)
!       );
  End Component psl_accel;
  
  Component base_img
***************
*** 447,452 ****
--- 492,501 ----
    logic_crc_err_immediate <= "0";
    logic_crc_code <= X"0000000000";
  
+   -- DDR4 ties
+   c0_ddr4_dm_dbi_n_9 <= '1';                      -- Disable sampling of data bits 79-72
+   c0_ddr4_ten        <= '0';                      -- Disable DDR4 connectivity test mode
+
       a0: psl_accel
        PORT MAP (
           ah_cvalid => a0h_cvalid,
***************
*** 504,510 ****
           ha_jeapar => ha0_jeapar,
           ha_jcompar => ha0_jcompar,
           ah_paren => a0h_paren,
!          ha_pclock => ha0_pclock
      );
  
      gold_factory <= '0'; --set to 1 to indicate images meant for user location. set to 0 to indicate factory image.
--- 553,579 ----
           ha_jeapar => ha0_jeapar,
           ha_jcompar => ha0_jcompar,
           ah_paren => a0h_paren,
!          ha_pclock => ha0_pclock,
!
!        refclk200_p        => refclk200_p,
!        refclk200_n        => refclk200_n,
!        c0_sys_clk_p       => c0_sys_clk_p,
!        c0_sys_clk_n       => c0_sys_clk_n,
! -- TODO:       dbg_bus            => dbg_bus,
!        c0_ddr4_adr        => c0_ddr4_adr,
!        c0_ddr4_ba         => c0_ddr4_ba,
!        c0_ddr4_cke        => c0_ddr4_cke,
!        c0_ddr4_cs_n       => c0_ddr4_cs_n,
!        c0_ddr4_dm_dbi_n   => c0_ddr4_dm_dbi_n,
!        c0_ddr4_dq         => c0_ddr4_dq,
!        c0_ddr4_dqs_c      => c0_ddr4_dqs_c,
!        c0_ddr4_dqs_t      => c0_ddr4_dqs_t,
!        c0_ddr4_odt        => c0_ddr4_odt,
!        c0_ddr4_bg         => c0_ddr4_bg,
!        c0_ddr4_reset_n    => c0_ddr4_reset_n,
!        c0_ddr4_act_n      => c0_ddr4_act_n,
!        c0_ddr4_ck_c       => c0_ddr4_ck_c,
!        c0_ddr4_ck_t       => c0_ddr4_ck_t
      );
  
      gold_factory <= '0'; --set to 1 to indicate images meant for user location. set to 0 to indicate factory image.
