*** ./Sources/top/psl_fpga.vhdl 2016-07-29 10:30:30.000000002 +0200
--- ./Sources/top/psl_fpga.vhdl 2016-10-12 08:20:34.000000002 +0200
***************
*** 59,66 ****
         pci0_o_txn_out6: out std_logic;
         pci0_o_txp_out7: out std_logic;
         pci0_o_txn_out7: out std_logic;
!        o_debug: out std_logic_vector(0 to 3));
  
         attribute secure_config : string;
         attribute secure_config of psl_fpga : entity is "PROTECT";
         attribute secure_netlist : string;       
--- 59,107 ----
         pci0_o_txn_out6: out std_logic;
         pci0_o_txp_out7: out std_logic;
         pci0_o_txn_out7: out std_logic;
!        o_debug: out std_logic_vector(0 to 3);
  
+        -- Alpha-Data KU3 card has two DDR3 SO-DIMMs
+        -- connected to Xilinx memory controllers
+ --       user_sw_l          : in    std_logic;
+ --       led_l              : out   std_logic_vector(2 downto 0);
+ 
+        c0_sys_clk_p       : IN STD_LOGIC;
+        c0_sys_clk_n       : IN STD_LOGIC;
+        c0_ddr3_addr       : out   std_logic_vector(15 downto 0);
+        c0_ddr3_ba         : out   std_logic_vector(2 downto 0);
+        c0_ddr3_ras_n      : out   std_logic;
+        c0_ddr3_cas_n      : out   std_logic;
+        c0_ddr3_reset_n    : out   std_logic;
+        c0_ddr3_cs_n       : out   std_logic_vector(1 downto 0);
+        c0_ddr3_cke        : out   std_logic_vector(1 downto 0);
+        c0_ddr3_ck_p       : out   std_logic_vector(1 downto 0);
+        c0_ddr3_ck_n       : out   std_logic_vector(1 downto 0);
+        c0_ddr3_we_n       : out   std_logic;
+        c0_ddr3_dm         : out   std_logic_vector(8 downto 0);
+        c0_ddr3_dq         : inout std_logic_vector(71 downto 0);
+        c0_ddr3_dqs_p      : inout std_logic_vector(8 downto 0);
+        c0_ddr3_dqs_n      : inout std_logic_vector(8 downto 0);
+        c0_ddr3_odt        : out   std_logic_vector(1 downto 0);
+ 
+        c1_sys_clk_p       : IN STD_LOGIC;
+        c1_sys_clk_n       : IN STD_LOGIC;
+        c1_ddr3_addr       : out   std_logic_vector(15 downto 0);
+        c1_ddr3_ba         : out   std_logic_vector(2 downto 0);
+        c1_ddr3_ras_n      : out   std_logic;
+        c1_ddr3_cas_n      : out   std_logic;
+        c1_ddr3_reset_n    : out   std_logic;
+        c1_ddr3_cs_n       : out   std_logic_vector(1 downto 0);
+        c1_ddr3_cke        : out   std_logic_vector(1 downto 0);
+        c1_ddr3_ck_p       : out   std_logic_vector(1 downto 0);
+        c1_ddr3_ck_n       : out   std_logic_vector(1 downto 0);
+        c1_ddr3_we_n       : out   std_logic;  
+        c1_ddr3_dm         : out   std_logic_vector(8 downto 0);
+        c1_ddr3_dq         : inout std_logic_vector(71 downto 0);
+        c1_ddr3_dqs_p      : inout std_logic_vector(8 downto 0);
+        c1_ddr3_dqs_n      : inout std_logic_vector(8 downto 0);
+        c1_ddr3_odt        : out   std_logic_vector(1 downto 0)
+        );
         attribute secure_config : string;
         attribute secure_config of psl_fpga : entity is "PROTECT";
         attribute secure_netlist : string;       
***************
*** 143,149 ****
         ha_jeapar: in std_ulogic;
         ha_jcompar: in std_ulogic;
         ah_paren: out std_ulogic;
!        ha_pclock: in std_ulogic);
  End Component psl_accel;
  
  Component base_img
--- 182,229 ----
         ha_jeapar: in std_ulogic;
         ha_jcompar: in std_ulogic;
         ah_paren: out std_ulogic;
!        ha_pclock: in std_ulogic;
! 
!        -- Alpha-Data KU3 card has two DDR3 SO-DIMMs
!        -- connected to Xilinx memory controllers
! --       user_sw_l          : in    std_logic;
! --       led_l              : out   std_logic_vector(2 downto 0);
!        c0_sys_clk_p       : IN STD_LOGIC;
!        c0_sys_clk_n       : IN STD_LOGIC;
!        c0_ddr3_addr       : out   std_logic_vector(15 downto 0);
!        c0_ddr3_ba         : out   std_logic_vector(2 downto 0);
!        c0_ddr3_ras_n      : out   std_logic;
!        c0_ddr3_cas_n      : out   std_logic;
!        c0_ddr3_reset_n    : out   std_logic;
!        c0_ddr3_cs_n       : out   std_logic_vector(1 downto 0);
!        c0_ddr3_cke        : out   std_logic_vector(1 downto 0);
!        c0_ddr3_ck_p       : out   std_logic_vector(1 downto 0);
!        c0_ddr3_ck_n       : out   std_logic_vector(1 downto 0);
!        c0_ddr3_we_n       : out   std_logic;
!        c0_ddr3_dm         : out   std_logic_vector(8 downto 0);
!        c0_ddr3_dq         : inout std_logic_vector(71 downto 0);
!        c0_ddr3_dqs_p      : inout std_logic_vector(8 downto 0);
!        c0_ddr3_dqs_n      : inout std_logic_vector(8 downto 0);
!        c0_ddr3_odt        : out   std_logic_vector(1 downto 0);
! 
!        c1_sys_clk_p       : IN STD_LOGIC;
!        c1_sys_clk_n       : IN STD_LOGIC;
!        c1_ddr3_addr       : out   std_logic_vector(15 downto 0);
!        c1_ddr3_ba         : out   std_logic_vector(2 downto 0);
!        c1_ddr3_ras_n      : out   std_logic;
!        c1_ddr3_cas_n      : out   std_logic;
!        c1_ddr3_reset_n    : out   std_logic;
!        c1_ddr3_cs_n       : out   std_logic_vector(1 downto 0);
!        c1_ddr3_cke        : out   std_logic_vector(1 downto 0);
!        c1_ddr3_ck_p       : out   std_logic_vector(1 downto 0);
!        c1_ddr3_ck_n       : out   std_logic_vector(1 downto 0);
!        c1_ddr3_we_n       : out   std_logic;  
!        c1_ddr3_dm         : out   std_logic_vector(8 downto 0);
!        c1_ddr3_dq         : inout std_logic_vector(71 downto 0);
!        c1_ddr3_dqs_p      : inout std_logic_vector(8 downto 0);
!        c1_ddr3_dqs_n      : inout std_logic_vector(8 downto 0);
!        c1_ddr3_odt        : out   std_logic_vector(1 downto 0)
!        );
  End Component psl_accel;
  
  Component base_img
***************
*** 377,383 ****
           ha_jeapar => ha0_jeapar,
           ha_jcompar => ha0_jcompar,
           ah_paren => a0h_paren,
!          ha_pclock => ha0_pclock
      );
  
      gold_factory <= '1'; --set to 1 to indicate images meant for user location. set to 0 to indicate factory image.
--- 455,497 ----
           ha_jeapar => ha0_jeapar,
           ha_jcompar => ha0_jcompar,
           ah_paren => a0h_paren,
!          ha_pclock => ha0_pclock,
! 
!          c0_sys_clk_p    => c0_sys_clk_p,
!          c0_sys_clk_n    => c0_sys_clk_n,
!          c0_ddr3_addr    => c0_ddr3_addr,
!          c0_ddr3_ba      => c0_ddr3_ba,
!          c0_ddr3_ras_n   => c0_ddr3_ras_n,
!          c0_ddr3_cas_n   => c0_ddr3_cas_n,
!          c0_ddr3_reset_n => c0_ddr3_reset_n,
!          c0_ddr3_cs_n    => c0_ddr3_cs_n,
!          c0_ddr3_cke     => c0_ddr3_cke,
!          c0_ddr3_ck_p    => c0_ddr3_ck_p,
!          c0_ddr3_ck_n    => c0_ddr3_ck_n,
!          c0_ddr3_we_n    => c0_ddr3_we_n,
!          c0_ddr3_dm      => c0_ddr3_dm, 
!          c0_ddr3_dq      => c0_ddr3_dq,
!          c0_ddr3_dqs_p   => c0_ddr3_dqs_p,
!          c0_ddr3_dqs_n   => c0_ddr3_dqs_n,
!          c0_ddr3_odt     => c0_ddr3_odt,
! 
!          c1_sys_clk_p    => c1_sys_clk_p,
!          c1_sys_clk_n    => c1_sys_clk_n,
!          c1_ddr3_addr    => c1_ddr3_addr,
!          c1_ddr3_ba      => c1_ddr3_ba,
!          c1_ddr3_ras_n   => c1_ddr3_ras_n,
!          c1_ddr3_cas_n   => c1_ddr3_cas_n,
!          c1_ddr3_reset_n => c1_ddr3_reset_n,
!          c1_ddr3_cs_n    => c1_ddr3_cs_n,
!          c1_ddr3_cke     => c1_ddr3_cke,
!          c1_ddr3_ck_p    => c1_ddr3_ck_p,
!          c1_ddr3_ck_n    => c1_ddr3_ck_n,
!          c1_ddr3_we_n    => c1_ddr3_we_n,
!          c1_ddr3_dm      => c1_ddr3_dm, 
!          c1_ddr3_dq      => c1_ddr3_dq,
!          c1_ddr3_dqs_p   => c1_ddr3_dqs_p,
!          c1_ddr3_dqs_n   => c1_ddr3_dqs_n,
!          c1_ddr3_odt     => c1_ddr3_odt
      );
  
      gold_factory <= '1'; --set to 1 to indicate images meant for user location. set to 0 to indicate factory image.
